### PR TITLE
add hook to dynamically configure repositories

### PR DESCRIPTION
**Pull request by evgeni**

### CODE REVIEW

**Review by ekohl — COMMENTED** [review]
Overall I like this, but this does need a Jenkins job change to benefit from this. Is there already such a PR?

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Add the right theforeman.org repositories
+
+set -eu
+
+if [ -z "${REPO_SETUP:-}" ]; then
```
Comment (ekohl) [inline puppet/modules/slave/files/pbuilder_F60addforemanrepo:6]: Should we make this fail hard if it isn't set? In our builds we basically always want it anyway. Perhaps it should be 
```suggestion
if [ "${REPO_SETUP:-}" = 0]; then
```

File: puppet/modules/slave/templates/pbuilder_F60addforemanrepo.erb [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+# Add the right theforeman.org repositories
+
+set -eu
+
+if [ -z "${REPO_SETUP:-}" ]; then
+  echo "Skipping automtic repository setup"
+  exit 0
+fi
+
+echo deb http://deb.theforeman.org/ <%= @$release %> ${REPO_VERSION:-nightly} > /etc/apt/sources.list.d/foreman.list
```
Comment (ekohl) [inline puppet/modules/slave/templates/pbuilder_F60addforemanrepo.erb:11]:
```suggestion
echo deb http://deb.theforeman.org/ <%= @release %> ${REPO_VERSION:-nightly} > /etc/apt/sources.list.d/foreman.list
```

**Review by ekohl — CHANGES_REQUESTED** [review]
Bikeshedding: `FOREMAN_REPO_VERSION`? Or just `FOREMAN_VERSION`?

Can we get rid of `REPO_SETUP`? Remove the default value for `REPO_VERSION` and deal with it that way?

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo (hunk 1 above)
Comment (evgeni) [reply puppet/modules/slave/files/pbuilder_F60addforemanrepo]: My idea was: don't fail hard *now*, allowing us to deploy this and still play around with the jobs and when we feel comfortable with the actual flow, just drop the `if` altogether.

Comment (ekohl) [reply puppet/modules/slave/files/pbuilder_F60addforemanrepo]: Can we really? The filename is the same as the one in the current jobs so it'll flip back and forth between a job and what Puppet applies.

**Comment by ekohl** [comment]
Should this include https://github.com/theforeman/foreman-infra/pull/1642 as well to make it complete?

File: puppet/modules/slave/files/pbuilder_F60addforemanrepo (hunk 1 above)
Comment (evgeni) [reply puppet/modules/slave/files/pbuilder_F60addforemanrepo]: implementation details ;)

File: puppet/modules/slave/templates/pbuilder_F60addforemanrepo.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+#!/bin/bash
+# Add the right theforeman.org repositories
+
+set -eu
+
+if [ -z "${FOREMAN_REPO_SETUP:-}" ]; then
```
Comment (ehelms) [inline puppet/modules/slave/templates/pbuilder_F60addforemanrepo.erb:6]: What's the case to skip this?

Comment (evgeni) [reply puppet/modules/slave/templates/pbuilder_F60addforemanrepo.erb:6]: https://github.com/theforeman/foreman-infra/pull/1643#discussion_r698433934

My idea was to easily switch between this and the "jenkins sets up things" for now, and remove later.

**Review by ekohl — APPROVED** [review]

**Review by ehelms — APPROVED** [review]

**Pull request merged**